sopel/config/types.py

```diff
@@ -111,27 +112,49 @@ class BaseValidated(object):
     :param str name: the attribute name to use in the config file
     :param default: the value to be returned if the setting has no value
                     (optional; defaults to :obj:`None`)
-    :type default: str
+    :type default: mixed
+    :param bool is_secret: tell if the attribute is secret/a password
+                           (optional; defaults to ``False``)
 
     ``default`` also can be set to :const:`sopel.config.types.NO_DEFAULT`, if
     the value *must* be configured by the user (i.e. there is no suitable
     default value). Trying to read an empty ``NO_DEFAULT`` value will raise
     :class:`AttributeError`.
     """
-    def __init__(self, name, default=None):
+    def __init__(self, name, default=None, is_secret=False):
         self.name = name
+        """Name of the attribute."""
         self.default = default
+        """Default value for this attribute. (default to ``None``)"""
+        self.is_secret = bool(is_secret)
+        """Tell if the attribute is secret/a password. (default to ``False``)
 
-    def configure(self, prompt, default, parent, section_name):
+        Sopel's configuration can contains password, secret key, private
+        information that must be treated as sensitive data. Such options should
+        be marked as "secret" with this attribute.
         """
-        With the ``prompt`` and ``default``, parse and return a value from
-        terminal.
+
+    def configure(self, prompt, default, parent, section_name):
+        """Parse and return a value from user's input.
```

Comment:
This is rough, and I welcome further improvements. Ideally this would appear on all `configure` methods, but I suspect we can just define it once on `BaseValidated`, remove the docstrings on child classes, and rely on inheritance for Sphinx to pull it in?

```suggestion
        """Parse and return a value from user's input.

        :param str prompt: text to show the user
        :param mixed default: default value used if no input given
        :param parent: usually the parent Config object
        :type parent: :class:`~sopel.config.Config`
        :param str section_name: the name of the containing section
```

And yes: You can also say "let's do this in a future PR".

Reply:
Yup, it works!